src/core/utils/fs.cpp

```diff
@@ -112,57 +113,52 @@ QString Utils::Fs::folderName(const QString& file_path)
 }
 
 /**
- * Remove an empty folder tree.
- *
- * This function will also remove .DS_Store files on Mac OS and
- * Thumbs.db on Windows.
+ * This function will first remove system cache files, e.g. `Thumbs.db`,
+ * `.DS_Store`. Then will try to remove the whole tree if the tree consist
+ * only of folders
  */
-bool Utils::Fs::smartRemoveEmptyFolderTree(const QString& dir_path)
+bool Utils::Fs::smartRemoveEmptyFolderTree(const QString& path)
 {
-    qDebug() << Q_FUNC_INFO << dir_path;
-    if (dir_path.isEmpty())
+    if (!QDir(path).exists())
         return false;
 
-    QDir dir(dir_path);
-    if (!dir.exists())
-        return true;
+    static const QStringList deleteFilesList = {
```

Comment:
I think in the future it will help if you structure it this way:
```c++
static const QStringList deleteFilesList = {
    // Windows
    "Thumbs.db",
    "desktop.ini",
    // Linux
    ".directory",
    // Mac OS
    ".DS_Store"
};
```